src/silly_sprites/opengl/buffer_object.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "../types.hpp"
+#include <glad/gl.h>
+#include <vector>
+
+namespace sly::gl {
+    class BufferObject final {
+    private:
+        using Points = std::vector<float>;
+        using Indices = std::vector<unsigned int>; // no output when in use usize so i stick to unsigned int
+
+        GLuint m_vao, m_vbo, m_ebo;
+
+        Points m_points;
+        Indices m_indices;
```

Comment:
Should we really store this data on the CPU side as well? The data is submitted to the GPU and also held as a copy by the CPU. I don't think that retrieving the contents of the buffer is a common use-case. And if it's really needed, you could query the contents of the GPU's buffer (which is really slow, yes, but I don't think we will need this very often if we need it at all).